packages/dashboard/src/layout/LayoutUtils.ts

```diff
@@ -574,7 +574,11 @@ class LayoutUtils {
    * @param root The GoldenLayout root to search and close the component in
    * @param config The GoldenLayout component config definition to close, eg. { component: 'IrisGridPanel', id: 'table-t' }
    */
-  static closeComponent(root: ContentItem, config: LayoutConfig): void {
+  static closeComponent(
+    root: ContentItem,
+    config: LayoutConfig,
+    isOverwrite = false
```

Comment:
You'll need to update the JS Docs to include this parameter... but before you do that, I suggest we change this to be an options object instead.
`isOverwrite` doesn't really make sense from the context of `LayoutUtils` - it doesn't (and shouldn't) know that the panel it is closing is a notebook panel type, or whatever. It doesn't care.

In fact, the way we're overriding the `close` behaviour has a bunch of hacks:
- Below where we're casting to unknown and then redefining close instead of just using the `ItemContainer` type
- We have to re-assign the `glContainer.close` property in `NoteboolPanel`, which is pretty hacky
  - That method returns `true` for seemingly no reason at all

We may as well clean this up while we're in here messing with it. My suggestions:
- Add a type `CloseOptions = { force?: boolean }` type to `ItemContainer`
- Add a `beforeClose` method that takes a callback or `null` to register a before close handler, e.g. `beforeClose(callback: ((options?: CloseOptions) => boolean) | null)`
- In the `close` method, redefine to take `CloseOptions` as an optional parameter, then check if there is a `beforeClose` callback registered, and if so then await the result of that, and only close if that resolves to success
- `NotebookPanel` then calls `beforeClose` instead of re-assigning `glContainer.close`

Let me know if you have questions. We could also make the `beforeClose` callback return a `Promise` instead for async, but I don't think we need that right now.